task test-ci fails

```
task: [test-ci] go test -tags=ci -cover -coverprofile=coverage.out -v ./...
        gocommender/cmd/server          coverage: 0.0% of statements
        gocommender/cmd/test-recommendations            coverage: 0.0% of statements
=== RUN   TestMockServicesIntegration
=== RUN   TestMockServicesIntegration/Mock_services_basic_functionality
--- PASS: TestMockServicesIntegration (0.00s)
    --- PASS: TestMockServicesIntegration/Mock_services_basic_functionality (0.00s)
=== RUN   TestCacheManagerIntegration
=== RUN   TestCacheManagerIntegration/Store_and_retrieve_artist
=== RUN   TestCacheManagerIntegration/Cache_expiry_logic
    integration_test.go:89: Expected condition to be true
=== RUN   TestCacheManagerIntegration/Bulk_cache_operations
=== RUN   TestCacheManagerIntegration/Cache_statistics
=== RUN   TestCacheManagerIntegration/Cleanup_expired_entries
    integration_test.go:139: Expected 0 >= 1
    integration_test.go:144: Expected value to be nil, got: &{cleanup-test-mbid Test Artist map[] 0   []   {   } 2025-07-20 09:21:35.132821 +0300 EEST 2025-07-27 09:21:35.132821 +0300 EEST}
    integration_test.go:145: Expected condition to be true
--- FAIL: TestCacheManagerIntegration (0.01s)
    --- PASS: TestCacheManagerIntegration/Store_and_retrieve_artist (0.00s)
    --- FAIL: TestCacheManagerIntegration/Cache_expiry_logic (0.00s)
    --- PASS: TestCacheManagerIntegration/Bulk_cache_operations (0.00s)
    --- PASS: TestCacheManagerIntegration/Cache_statistics (0.00s)
    --- FAIL: TestCacheManagerIntegration/Cleanup_expired_entries (0.00s)
=== RUN   TestEnrichmentServiceIntegration
=== RUN   TestEnrichmentServiceIntegration/Mock_enrichment_service_functionality
--- PASS: TestEnrichmentServiceIntegration (0.00s)
    --- PASS: TestEnrichmentServiceIntegration/Mock_enrichment_service_functionality (0.00s)
=== RUN   TestAPIServerIntegration
=== RUN   TestAPIServerIntegration/Mock_services_compatibility
--- PASS: TestAPIServerIntegration (0.00s)
    --- PASS: TestAPIServerIntegration/Mock_services_compatibility (0.00s)
=== RUN   TestConcurrentOperations
=== RUN   TestConcurrentOperations/Concurrent_cache_operations
    integration_test.go:254: Goroutine error: SQL logic error: no such table: artists (1)
    integration_test.go:254: Goroutine error: SQL logic error: no such table: artists (1)
    integration_test.go:254: Goroutine error: SQL logic error: no such table: artists (1)
    integration_test.go:254: Goroutine error: SQL logic error: no such table: artists (1)
    integration_test.go:254: Goroutine error: SQL logic error: no such table: artists (1)
    integration_test.go:254: Goroutine error: SQL logic error: no such table: artists (1)
    integration_test.go:254: Goroutine error: failed to check cache expiry: failed to check expiry: SQL logic error: no such table: artists (1)
    integration_test.go:254: Goroutine error: SQL logic error: no such table: artists (1)
    integration_test.go:254: Goroutine error: SQL logic error: no such table: artists (1)
    integration_test.go:254: Goroutine error: SQL logic error: no such table: artists (1)
    integration_test.go:260: Had 10 errors in concurrent operations
    integration_test.go:265: Expected no error, got: failed to get total count: SQL logic error: no such table: artists (1)
--- FAIL: TestConcurrentOperations (0.00s)
    --- FAIL: TestConcurrentOperations/Concurrent_cache_operations (0.00s)
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -117,7 +117,7 @@ jobs:
             type=semver,pattern={{version}}
             type=semver,pattern={{major}}.{{minor}}
             type=semver,pattern={{major}}
-            type=sha,prefix={{branch}}-
+            type=sha,enable={{is_default_branch}}
             type=raw,value=latest,enable={{is_default_branch}}
           labels: |
             org.opencontainers.image.title=GoCommender

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -61,10 +61,13 @@ jobs:
         with:
           submodules: recursive
 
-      - name: Run Gosec Security Scanner
-        uses: securecodewarrior/github-action-gosec@master
+      - name: Set up Go
+        uses: actions/setup-go@v4
         with:
-          args: "-severity medium ./..."
+          go-version: "1.24.5"
+
+      - name: Run Go Vulnerability Check
+        uses: golang/govulncheck-action@v1
 
       - name: Run Trivy vulnerability scanner on filesystem
         uses: aquasecurity/trivy-action@master

--- a/internal/api/integration_test.go
+++ b/internal/api/integration_test.go
@@ -76,10 +76,13 @@ func TestCacheManagerIntegration(t *testing.T) {
 	t.Run("Cache expiry logic", func(t *testing.T) {
 		artist := testutil.TestArtistExpired()
 
-		// Manually set cache expiry to past
-		artist.CacheExpiry = time.Now().Add(-time.Hour)
-
+		// First cache the artist normally
 		err := cacheManager.CacheArtist(artist, config)
+		testutil.AssertNoError(t, err)
+
+		// Then manually set cache expiry to past to simulate expiry
+		expiredTime := time.Now().Add(-time.Hour)
+		err = cacheManager.UpdateCacheExpiry(artist.MBID, expiredTime)
 		testutil.AssertNoError(t, err)
 
 		// Should indicate needs fetch due to expiry
@@ -128,9 +131,14 @@ func TestCacheManagerIntegration(t *testing.T) {
 		// Create an artist that's been expired for a long time
 		artist := testutil.TestArtistMinimal()
 		artist.MBID = "cleanup-test-mbid"
-		artist.CacheExpiry = time.Now().Add(-2 * time.Hour) // Expired 2 hours ago
 
+		// First cache the artist normally
 		err := cacheManager.CacheArtist(artist, config)
+		testutil.AssertNoError(t, err)
+
+		// Then manually set cache expiry to past to simulate long expiry
+		expiredTime := time.Now().Add(-2 * time.Hour) // Expired 2 hours ago
+		err = cacheManager.UpdateCacheExpiry(artist.MBID, expiredTime)
 		testutil.AssertNoError(t, err)
 
 		// Cleanup entries that have been expired for more than 1 hour
@@ -141,7 +149,10 @@ func TestCacheManagerIntegration(t *testing.T) {
 		// Verify the artist was deleted
 		retrieved, needsFetch, err := cacheManager.GetOrFetchArtist(artist.MBID)
 		testutil.AssertNoError(t, err)
-		testutil.AssertNil(t, retrieved)
+		if retrieved != nil {
+			t.Errorf("Expected artist to be deleted, but found: MBID=%s, Name=%s, CacheExpiry=%v",
+				retrieved.MBID, retrieved.Name, retrieved.CacheExpiry)
+		}
 		testutil.AssertTrue(t, needsFetch)
 	})
 }

--- a/internal/testutil/helpers.go
+++ b/internal/testutil/helpers.go
@@ -9,10 +9,15 @@ import (
 
 // CreateTestDB creates an in-memory SQLite database for testing
 func CreateTestDB(t *testing.T) (*sql.DB, func()) {
-	db, err := sql.Open("sqlite", ":memory:")
+	// Use shared cache in-memory database for concurrent access
+	db, err := sql.Open("sqlite", ":memory:?cache=shared")
 	if err != nil {
 		t.Fatalf("Failed to create test database: %v", err)
 	}
+
+	// Set connection pool to use single connection to avoid race conditions
+	db.SetMaxOpenConns(1)
+	db.SetMaxIdleConns(1)
 
 	// Create schema
 	schema := `

